From 778e931842047e621220f8c3b816e9aa68161e66 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 31 May 2021 20:44:58 +0300
Subject: [PATCH] [SW-0] lkd: fix compilation issue

There is a compiration error:
-----------------------cut------------------------
preboot/module_lkd.c: In function ‘module_lkd_run’:
preboot/module_lkd.c:502:4: error: ‘offset’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  502 |    module_lkd_send_status(COMMS_STS_OK,
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  503 |             COMMS_SRAM, offset);
      |             ~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [preboot/module_lkd.o] Error 1
-----------------------cut------------------------

Fix this by initializing the variable.

Change-Id: Ib54a32646c8efc7717df2a90cfd12d9ce78e2acd
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 9031925..b2402d0 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -473,7 +473,7 @@
 void module_lkd_run(void)
 {
 	int err = 0;
-	u32 offset;
+	u32 offset = 0;
 
 	if (is_legacy_comms(sts.cmd.val))
 		return module_lkd_run_legacy();
